From 3c303d4cf98bfdafdb933e6313acd189352dd0d5 Mon Sep 17 00:00:00 2001
From: Joel Pinto <joelpostman97@gmail.com>
Date: Sat, 5 Mar 2022 23:38:13 +0000
Subject: [PATCH] kernel-src: Specify the gpio-lines-name property

To identify the gpios base on their names when using the libgpiod tools,
the gpio-line-names property must be defined, for such, this commit
implements a device tree overlay to describe this desired hardware property.

Signed-off-by: Joel Pinto <joelpostman97@gmail.com>
---
 arch/arm/boot/dts/overlays/Makefile           |  1 +
 .../gpio-line-names-bcm2709-overlay.dts       | 68 +++++++++++++++++++
 2 files changed, 69 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/gpio-line-names-bcm2709-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index ab9fa47c8876..2c0c831ab4d3 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -52,6 +52,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	gpio-ir.dtbo \
 	gpio-ir-tx.dtbo \
 	gpio-key.dtbo \
+	gpio-line-names-bcm2709.dtbo \
 	gpio-no-bank0-irq.dtbo \
 	gpio-no-irq.dtbo \
 	gpio-poweroff.dtbo \
diff --git a/arch/arm/boot/dts/overlays/gpio-line-names-bcm2709-overlay.dts b/arch/arm/boot/dts/overlays/gpio-line-names-bcm2709-overlay.dts
new file mode 100644
index 000000000000..d689600c5051
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/gpio-line-names-bcm2709-overlay.dts
@@ -0,0 +1,68 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+        compatible = "brcm, bcm2836";
+
+        fragment@0 {
+                target = <&gpio>;
+                __overlay__ {
+                        gpio-line-names = "ID_SDA",
+                          "ID_SCL",
+                          "SDA1",
+                          "SCL1",
+                          "GPIO_GCLK",
+                          "GPIO5",
+                          "GPIO6",
+                          "SPI_CE1_N",
+                          "SPI_CE0_N",
+                          "SPI_MISO",
+                          "SPI_MOSI",
+                          "SPI_SCLK",
+                          "GPIO12",
+                          "GPIO13",
+                          /* Serial port */
+                          "TXD0",
+                          "RXD0",
+                          "GPIO16",
+                          "GPIO17",
+                          "GPIO18",
+                          "GPIO19",
+                          "GPIO20",
+                          "GPIO21",
+                          "GPIO22",
+                          "GPIO23",
+                          "GPIO24",
+                          "GPIO25",
+                          "GPIO26",
+                          "GPIO27",
+                          "SDA0",
+                          "SCL0",
+                          "NC", /* GPIO30 */
+                          "LAN_RUN",
+                          "CAM_GPIO1",
+                          "NC", /* GPIO33 */
+                          "NC", /* GPIO34 */
+                          "PWR_LOW_N",
+                          "NC", /* GPIO36 */
+                          "NC", /* GPIO37 */
+                          "USB_LIMIT",
+                          "NC", /* GPIO39 */
+                          "PWM0_OUT",
+                          "CAM_GPIO0",
+                          "SMPS_SCL",
+                          "SMPS_SDA",
+                          "ETH_CLK",
+                          "PWM1_OUT",
+                          "HDMI_HPD_N",
+                          "STATUS_LED",
+                          /* Used by SD Card */
+                          "SD_CLK_R",
+                          "SD_CMD_R",
+                          "SD_DATA0_R",
+                          "SD_DATA1_R",
+                          "SD_DATA2_R",
+                          "SD_DATA3_R";
+                };
+        };
+};
-- 
2.17.1

